backend/gsr_booking/api_wrapper.py

```diff
@@ -31,175 +37,66 @@ class CreditType(Enum):
-        penn_labs = Group.objects.get(name="Penn Labs")
-        me_group = Group.objects.get(name="Me", owner=user)
-        membership = GroupMembership.objects.filter(group=me_group).first()
-        return membership.check_wharton() or user in penn_labs.members.all()
-
-    def book_room(self, gid, rid, room_name, start, end, user, group_book=None):
-        gsr = get_object_or_404(GSR, gid=gid)
-
-        # error catching on view side
-        if gsr.kind == GSR.KIND_WHARTON:
-            booking_id = self.WLW.book_room(rid, start, end, user, gsr.lid).get("booking_id")
-        else:
-            booking_id = self.LCW.book_room(rid, start, end, user).get("booking_id")
-
-        # creates booking on database
-        # TODO: break start / end time into smaller chunks and pool credit for group booking
-        booking = GSRBooking.objects.create(
-            user=user,
-            booking_id=str(booking_id),
-            gsr=gsr,
-            room_id=rid,
-            room_name=room_name,
-            start=datetime.datetime.strptime(start, "%Y-%m-%dT%H:%M:%S%z"),
-            end=datetime.datetime.strptime(end, "%Y-%m-%dT%H:%M:%S%z"),
-        )
-
-        # create reservation with single-person-group containing user
-        # TODO: create reservation with group that frontend passes in
-        if not group_book:
-            single_person_group = get_object_or_404(Group, owner=user)
-            reservation = Reservation.objects.create(
-                start=start, end=end, creator=user, group=single_person_group
-            )
-            booking.reservation = reservation
-            booking.save()
-
-        return booking
-
-    def get_availability(self, lid, gid, start, end, user):
-        # checks which GSR class to use
-        gsr = get_object_or_404(GSR, gid=gid)
-
-        if gsr.kind == GSR.KIND_WHARTON:
-            rooms = self.WLW.get_availability(lid, start, end, user.username)
-            return {"name": gsr.name, "gid": gsr.gid, "rooms": rooms}
-        else:
-            rooms = self.LCW.get_availability(lid, start, end)
-            # cleans data to match Wharton wrapper
-            try:
-                lc_gsr = [x for x in rooms["categories"] if x["cid"] == int(gid)][0]
-            except IndexError:
-                raise APIError("Unknown GSR")
-
-            for room in lc_gsr["rooms"]:
-                for availability in room["availability"]:
-                    availability["start_time"] = availability["from"]
-                    availability["end_time"] = availability["to"]
-                    del availability["from"]
-                    del availability["to"]
-
-            context = {"name": lc_gsr["name"], "gid": lc_gsr["cid"]}
-            context["rooms"] = [
-                {"room_name": x["name"], "id": x["id"], "availability": x["availability"]}
-                for x in lc_gsr["rooms"]
-            ]
-            return context
-
+class AbstractBookingWrapper(ABC):
```

Comment:
holy cow, i haven't seen a Python abstract class in a while LOL

Reply:
nit: for all the methods, raise a `NotImplemented` exception instead of `pass`, unless this is just like a partial interface?